packages/framework-provider-aws/test/library/searcher-adapter.test.ts

```diff
@@ -0,0 +1,202 @@
+  describe('The "searchReadModel" method', () => {
+    const config: BoosterConfig = new BoosterConfig('test')
+    const logger: Logger = {
+      info: fake(),
+      error: fake(),
+      debug: fake(),
+    }
+    const readModelName: string = random.word()
+
+    let database: SinonStubbedInstance<DynamoDB.DocumentClient>
+    const expectedParams = {
+      TableName: config.resourceNames.forReadModel(readModelName),
+      ConsistentRead: true,
+    }
+
+    class Money {
+      constructor(public cents: number, public currency: string) {}
+    }
+
+    class Item {
+      constructor(public sku: string, public price: Money) {}
+    }
+
+    class Product {
+      constructor(
+        readonly id: string,
+        readonly stock: number,
+        public mainItem: Item,
+        public items: Array<Item>,
+        public buyers: Array<string>,
+        public days: Array<number>,
+        public pairs: Array<Array<number>>
+      ) {}
+    }
+
+    beforeEach(() => {
+      database = createStubInstance(DynamoDB.DocumentClient, {
+        scan: {
+          promise: stub().returns({
+            result: stub().returns({}),
+          }),
+        } as any,
+      })
+    })
+    after(() => {
+      restore()
+    })
+
+    it('Executes query without filters', async () => {
+      const result = await searchReadModel(database, config, logger, readModelName, {})
+
+      expect(database.scan).to.have.been.calledWithExactly(expectedParams)
+      expect(result).to.be.deep.equal([])
+    })
+
+    it('Executes query simple filters', async () => {
+      const expectedInput = {
+        ...expectedParams,
+        FilterExpression: [
+          'contains(#id, :id_0)',
+          'AND #id IN (:id_1_0,:id_1_1,:id_1_2)',
+          'AND #stock > :stock_0',
+          'AND #stock <= :stock_1',
+        ].join(' '),
+        ExpressionAttributeNames: { '#id': 'id', '#stock': 'stock' },
+        ExpressionAttributeValues: {
+          ':id_0': '3',
+          ':id_1_0': 'test1',
+          ':id_1_1': 'test2',
+          ':id_1_2': 'test3',
+          ':stock_0': 0,
+          ':stock_1': 10,
+        },
+      }
+      const filters: FilterFor<Product> = {
+        id: { contains: '3', in: ['test1', 'test2', 'test3'] },
+        stock: { gt: 0, lte: 10 },
+      }
+
+      await searchReadModel(database, config, logger, readModelName, filters as FilterFor<any>)
+
+      expect(database.scan).to.have.been.calledWithExactly(expectedInput)
+    })
+
+    it('Executes query using NOT in filters', async () => {
+      const expectedInput = {
+        ...expectedParams,
+        FilterExpression: 'contains(#id, :id_0) AND NOT (#id = :id_1)',
+        ExpressionAttributeNames: { '#id': 'id' },
+        ExpressionAttributeValues: {
+          ':id_0': '3',
+          ':id_1': '333',
+        },
+      }
+      const filters: FilterFor<Product> = {
+        id: { contains: '3' },
+        not: { id: { eq: '333' } },
+      }
```

Comment:
Pretty cool DSL. Let's make sure it's properly documented before merging this PR, otherwise nobody will use this feature!

Reply:
1000% agree. Now is the time to document this thoroughly

Reply:
I think it would make more sense to document it on the GraphQL new schema branch.

Reply:
Well, you have to document two things: how you use it from a handler and how you'll use it when it's available from the API. I'd add the docs on each PR to make sure it's not forgotten later

Reply:
Wouldn't be enough with the current `Getting and filtering read models data at code level` section I added on the docs?